### DISCUSSION TITLE
EACCES: permission denied since v39

### DISCUSSION THREAD

**Discussion by vicvytruve** [renovatebot/renovate#32399] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, with renovate/renovate docker image

### Please tell us more about your question or problem

Since the v39, we are experiencing issue with Renovate pipelines, they are all failing with this error : 
```
 {
           "name": "renovate",
           "level": 50,
           "logContext": "5Tqwka6OasD6Zsqu95p3A",
           "repository": "xx/yy",
           "err": {
             "errno": -13,
             "code": "EACCES",
             "syscall": "mkdir",
             "path": "/builds/xx/yy/renovate/renovate-runner/renovate/cache/__renovate-private-cache",
             "message": "EACCES: permission denied, mkdir '/builds/xx/yy/renovate/renovate-runner/renovate/cache/__renovate-private-cache'",
             "stack": "Error: EACCES: permission denied, mkdir '/builds/xx/yy/renovate/renovate-runner/renovate/cache/__renovate-private-cache'"
           },
           "msg": "Repository has unknown error"
         },
```

It's failing since this image : https://hub.docker.com/r/renovate/renovate/tags?name=5020894a1da353

Works again when using [38.142](https://hub.docker.com/layers/renovate/renovate/38.142/images/sha256-0036a01978a7ce235186920baff328a9d0aecfd34e3c335b5702169b0dc2da20?context=explore).

Any idea, should we update any kind of config ?

         
         

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
The image user ID changed from 1001 to 12021. You could try:
- Deleting/dropping cache temporarily, or
- Overriding user ID back to 1001

**Reply by neophob** [reply]
@rarkins I think I know what the problem is.. 

the local cache path contains the repo path - in my case the cache path is created like:

`/builds/RENOVATE-RUNNER-REPO-URL/cache/renovate/repository/REPO-URL-OF-PROJECT-THAT-IS-RENOVATED`

in both cases my repo url contains forward slashes `/` - which of course break the path.. It might have even a security implication...

To test I set `RENOVATE_CACHE_DIR: '/foo'` - and there I saw exact the same error message:

```
           "err": {
             "errno": -13,
             "code": "EACCES",
             "syscall": "mkdir",
             "path": "/foo",
             "message": "EACCES: permission denied, mkdir '/foo'",
             "stack": "Error: EACCES: permission denied, mkdir '/foo'"
           },
```

**Reply by rarkins** [reply]
The last part should be `cache/renovate/repository/gitlab/group/project.json`

**Reply by neophob** [reply]
I forgot to mention, when I set

`RENOVATE_CACHE_DIR: '/tmp'`

then the error does not happens, means if the renovate runner url contains a forward slash, then the error happens

**Reply by rarkins** [reply]
@neophob so when it doesn't work, what path do you think it's trying to use?

@Gabriel-Ladzaretti this seems closely related to what you're working on

**Comment by neophob** [comment]
@rarkins the failing path is `/builds/company/it/cin/stuff/name-renovatebot-job/renovate/cache/renovate/repository/gitlab/company/it/cos/stuff2` - the repo name is `company/it/cin/stuff/name-renovatebot-job` and I assume thats the problem. When I overwrite this path with `RENOVATE_CACHE_DIR` it works

just for the record, this is the complete error

```
           "err": {
             "errno": -13,
             "code": "EACCES",
             "syscall": "mkdir",
             "path": "/builds/company/it/cin/stuff/name-renovatebot-job/renovate/cache/renovate/repository/gitlab/company/it/cos/stuff2",
             "message": "EACCES: permission denied, mkdir '/builds/company/it/cin/stuff/name-renovatebot-job/renovate/cache/renovate/repository/gitlab/company/it/cos/stuff2'",
             "stack": "Error: EACCES: permission denied, mkdir '/builds/company/it/cin/stuff/name-renovatebot-job/renovate/cache/renovate/repository/gitlab/company/it/cos/stuff2'"
           },
```

**Reply by rarkins** [reply]
Possibly still related to change of image UID in v39

**Reply by neophob** [reply]
I thought this too, but we do not persist any cache across different renovate run - and to be extra sure I also deleted the runners cache for that job.. any ideas how to validate it might be the UID / permission issue?

**Reply by viceice** [reply]
did you tried to clean the cache after upgrading to renovate v39?

**Reply by neophob** [reply]
>did you tried to clean the cache after upgrading to renovate v39?

yes I did

**Reply by viceice** [reply]
how do you run renovate? please provide the workflow.

**Reply by neophob** [reply, marked as answer]
Sorry, my bad - it looks like I messed up that last run, indeed it works now, so @rarkins was right about the UID issue!

Thanks again for the quick fix and support guys!

**Reply by Sax388** [reply]
Thanks a ton for raising this question. The selected answer doesn't really show the resolution though.

I set `RENOVATE_CACHE_DIR` to `/tmp` and then it worked but when I switched back it didn't work again.

So the next thing I tried was to reset the cache ([for gitlab](https://docs.gitlab.com/ee/ci/caching/#clear-the-cache-manually)):
![image](https://github.com/user-attachments/assets/4f105bf7-b300-417d-a7c1-3ad6949a51f3)
![image](https://github.com/user-attachments/assets/35844dfc-a716-44d6-b4eb-ee171c9a90f1)
And this time it worked like a charm 🎉 .

Was it mentioned somewhere that the cache needed to be invalidated for this update? So perhaps next time I'm prepared 🙂. Thanks a lot for this awesome tool!

**Comment by vicvytruve** [comment]
It work for me after removing the cache. Thank you !

**Comment by fgreinacher** [comment]
Just cross-posting here: I made a suggestion to avoid this when using the `renovate-runner` template: https://gitlab.com/renovate-bot/renovate-runner/-/issues/108

**Reply by viceice** [reply]
It's documented to require to clear caches since the v19 release of the templates, but it seems nobody reads them 😞 

https://gitlab.com/renovate-bot/renovate-runner/-/releases/v19.0.0#-breaking-changes

**Reply by Sax388** [reply]
Thank you, I just changed my notification settings so I can read them accordingly.

**Reply by neophob** [reply]
@viceice I was looking https://docs.renovatebot.com/release-notes-for-major-versions/ here for breaking changes

**Reply by vicvytruve** [reply]
Same as @neophob , I was looking at this  documentation

**Reply by viceice** [reply]
@rarkins @secustor we probably should drop that page as long as there's no maintainer for it?

**Comment by gberche-orange** [comment]
When running renovate on a local desktop using docker, we worked around by override renovate user with current shell user which the following syntax

```bash
docker run \
    --rm \
    -u $(id -u):$(id -g) \
    -v ${CACHED_TMP_RENOVATE}:/tmp/renovate \
...
```

/CC @o-orand
